packages/backend-app-api/src/services/implementations/httpRouter/middleware/cookieMiddlewareFactory.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { getBearerTokenFromAuthorizationHeader } from '@backstage/plugin-auth-node';
+import { NextFunction, Request, Response } from 'express';
+import { decodeJwt } from 'jose';
+import { URL } from 'url';
+import lzstring from 'lz-string';
+import { Config } from '@backstage/config';
+
+function setTokenCookie(
+  res: Response,
+  options: { token: string; secure: boolean; cookieDomain: string },
+) {
+  try {
+    const payload = decodeJwt(options.token);
+    res.cookie('token', options.token, {
+      encode: lzstring.compressToEncodedURIComponent,
+      expires: new Date(payload.exp ? payload.exp * 1000 : 0),
+      secure: options.secure,
+      sameSite: 'lax',
```

Comment:
Can't we set this to `strict` or will that break in certain cases?
```suggestion
      sameSite: 'strict',
```

Reply:
This was pulled from the authenticate-api tutorial. I think the value in the tutorial is cross-linking from other sites. In the case of this PR though, I'm not convinced that will be part of the equation as we're not all in on cookie auth.